internal/pkg/levee/testdata/src/example.com/tests/colocation/tests.go

```diff
@@ -16,22 +16,138 @@ package colocation
+	core.Sink(i) // TODO want "a source has reached a sink"
+}
+
+func colocateUnsafePointer(core.Source, unsafe.Pointer) {}
+
+func TestUnsafePointerIsTainted(s core.Source, up unsafe.Pointer) {
+	colocateUnsafePointer(s, up)
+	core.Sink(up) // want "a source has reached a sink"
+}
+
+type PointerHolder struct{ ptr *core.Source }
+
+func colocatePointerHolder(core.Source, PointerHolder) {}
+
+func TestNamedStructPointerHolderIsTainted(s core.Source, ph PointerHolder) {
+	// This test is failing because ph is created by an Alloc,
+	// and we don't traverse through non-array Allocs
+	colocatePointerHolder(s, ph)
+	core.Sink(ph) // TODO want "a source has reached a sink"
+}
+
+func colocateArrOfValues(core.Source, [1]string) {}
+
+func TestArrOfValuesIsNotTainted(s core.Source, arr [1]string) {
+	colocateArrOfValues(s, arr)
+	core.Sink(arr)
+}
+
+func colocateArrOfPointers(core.Source, [1]*string) {}
+
+func TestArrOfPointersIsTainted(s core.Source, arr [1]*string) {
+	colocateArrOfPointers(s, arr)
+	core.Sink(arr) // want "a source has reached a sink"
+}
+
+func colocateFunc(core.Source, func()) {}
+
+func TestFuncIsNotTainted(s core.Source, f func()) {
+	colocateFunc(s, f)
+	core.Sink(f)
+}
+
+func colocateReferenceCollections(core.Source, map[string]string, chan string, []string) {}
+
+func TestReferenceCollectionsAreTainted(s core.Source) {
+	m := make(map[string]string)
+	c := make(chan string)
+	sl := make([]string, 0)
+	colocateReferenceCollections(s, m, c, sl)
+	core.Sink(m)  // want "a source has reached a sink"
+	core.Sink(c)  // want "a source has reached a sink"
+	core.Sink(sl) // want "a source has reached a sink"
 }
 
-func TestTaintIsPropagatedToColocatedPointerArgumentsThroughEface(s core.Source, i core.Innocuous, ip *core.Innocuous) {
-	taintColocatedEface(s, &i, ip)
-	core.Sink(s)  // want "a source has reached a sink"
-	core.Sink(i)  // TODO want "a source has reached a sink"
-	core.Sink(ip) // want "a source has reached a sink"
+func colocateReflectValue(core.Source, reflect.Value) {}
+
+func TestReflectValuesAreTainted(s core.Source, r reflect.Value) {
+	// This test is failing because ph is created by an Alloc,
+	// and we don't traverse through non-array Allocs
+	colocateReflectValue(s, r)
+	core.Sink(r) // TODO want "a source has reached a sink"
+}
+
+func colocateEface(s core.Source, taintees ...interface{}) {}
+
+func TestTaintedEface(s core.Source, i interface{}) {
+	colocateEface(s, i)
+	core.Sink(i) // want "a source has reached a sink"
+
+}
+
+func TestTaintedThroughEface(s core.Source, str string, i core.Innocuous) {
+	colocateEface(s, str, i)
+	// Ideally, we wouldn't want reports for either of those values, because they
+	// are not pointers.
+	// However, because they are passed to the call via an interface type,
+	// we have no easy way to know that these values can't actually be tainted.
+	core.Sink(str) // want "a source has reached a sink"
+	core.Sink(i)   // TODO want "a source has reached a sink"
+}
```

Comment:
Delegating to `pointer.CanPoint` and reverting your changes in handling `UnOp` and `Slice` yields this test and the below as passing.

Reply:
Hmm. That's not what I'm seeing. If I revert the change to `UnOp`, the `TestArrOfPointersIsTainted` test case fails. If I revert the change to `Slice`, `TestTaintedEface` and the first want in `TestTaintedThroughEface` both fail. Neither change seems to have any impact on the `TestPointerTaintedThroughEface` case, which is below `TestTaintedThroughEface`.

Reply:
Yeah, I think two-weeks-ago-Me got into a weird partial state.